packages/e2e-test-utils-playwright/src/metrics/index.ts

```diff
@@ -273,4 +289,95 @@ export class Metrics {
 			)
 			.map( ( item ) => ( item.dur ? item.dur / 1000 : 0 ) );
 	}
+
+	/**
+	 * Initializes the web-vitals library upon next page navigation.
+	 *
+	 * Defaults to automatically triggering the navigation,
+	 * but it can also be done manually.
+	 *
+	 * @example
+	 * ```js
+	 * await metrics.initWebVitals();
+	 * console.log( await metrics.getWebVitals() );
+	 * ```
+	 *
+	 * @example
+	 * ```js
+	 * await metrics.initWebVitals( false );
+	 * await page.goto( '/some-other-page' );
+	 * console.log( await metrics.getWebVitals() );
+	 * ```
+	 *
+	 * @param reload Whether to force navigation by reloading the current page.
+	 */
+	async initWebVitals( reload = true ) {
+		await this.page.addInitScript( {
+			path: join(
+				__dirname,
+				'../../../../node_modules/web-vitals/dist/web-vitals.umd.cjs'
+			),
```

Comment:
Couldn't we inject `web-vitals` as a `type="module"`, as recommended in https://github.com/GoogleChrome/web-vitals?

Reply:
We could, but I don't see a benefit in doing so in this context. Quite the opposite IMHO, having to add an inline script makes implementation less maintainable and loading the library from a CDN prevents offline usage.

Reply:
> having to add an inline script makes implementation less readable and maintainable.

I'm not sure I understand. Isn't the `window.addEventListener...` code you have below also added to the page as an inline script?

> And loading the library from a CDN prevents offline usage, so not preferred either.

Is it not possible to import from the local file in the module script?

Reply:
I meant specifically adding the module import as an inline script is a bit messy. addInitScript() is very clean on the other hand.

And no, not to my knowledge. You would need a server to serve the file somehow, and that is not available.

Reply:
Given the offline usage benefits of this approach I don't see why this want to change things.

It works, is readable, and reliable.

Reply:
> I meant specifically adding the module import as an inline script is a bit messy. addInitScript() is very clean on the other hand.

Isn't `addInitScript` technically adding an inline script under the hood? I'm not very familiar with Playwright, that's why I asked for clarification. I simply don't understand how those two are different :)

Reply:
Yes that‘s right, I should have clarified a bit more. I was thinking of how it‘s implemented in the benchmark-web-vitals script.

Reply:
`addInitScript` also [evaluates whenever a child frame attaches to the page](https://playwright.dev/docs/api/class-page#page-add-init-script), which in the context of WordPress unfortunately, could happen a lot when rendering a set of Pattern previews.

I wonder if what we intent here is to use [`addScriptTag`](https://playwright.dev/docs/api/class-page#page-add-init-script) instead?

Reply:
Is that so? I understood the docs there slightly differently, that it depends on which object you call the method on (i.e. `context.addInitScript()` or `page.addInitScript()`.
Given that web vitals usage is intended for the frontend and not the editor, I wouldn't see a big issue with that though. But feel free to submit a PR 👍

Reply:
A browser context can have many pages, and a page can have many frames. IIUC, `context.addInitScript()` will add the script to every opened page and any future page, `page.addInitScript()` will add the script to every frame and any future frame.

> Given that web vitals usage is intended for the frontend and not the editor

I see! It's not a big deal then 👍.